File: app/src/main/java/com/example/wellbees_project/MainActivity.java
package com.example.wellbees_project;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.util.Log;

import com.example.wellbees_project.allSources.AllSourcesAdapter;
import com.example.wellbees_project.allSources.Source;
import com.example.wellbees_project.allSources.SourcesReply;
import com.example.wellbees_project.detailedSources.Article;
import com.example.wellbees_project.detailedSources.DetailedSource;

import java.util.List;




import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = MainActivity.class.getSimpleName();

    private final static String API_KEY = "d8920f7f20be4311a9d4e2d76dc68139";
    private final static String COUNTRY = "us";

    private SourcesDAOInterface sourcesDAOInterface;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final RecyclerView recyclerView = (RecyclerView) findViewById(R.id.source_recyclerview);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        sourcesDAOInterface = ApiUtils.getSourcesDaoInterface();

        allSources(recyclerView);

    }


    public void allSources(RecyclerView recyclerView){
        sourcesDAOInterface.allSources(COUNTRY,API_KEY).enqueue(new Callback<SourcesReply>() {
            @Override
            public void onResponse(Call<SourcesReply> call, Response<SourcesReply> response) {

                List<Source> sourcesList = response.body().getSources();

                for (Source k:sourcesList) {
                    Log.e("**************", "****************");
                    Log.e("source id: ", k.getId());
                    Log.e("source id: ", k.getCountry());
                    Log.e("source id: ", k.getName());
                }

                recyclerView.setAdapter(new AllSourcesAdapter(sourcesList,R.layout.list_item,getApplicationContext()));


            }
            @Override
            public void onFailure(Call<SourcesReply> call, Throwable t) {
                Log.e(TAG,t.toString());
            }
        });
    }



}

File: app/src/main/java/com/example/wellbees_project/MainActivity.kt
package com.example.wellbees_project

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.provider.ContactsContract.CommonDataKinds.StructuredPostal.COUNTRY
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.wellbees_project.allSources.AllSourcesAdapter
import com.example.wellbees_project.allSources.Source
import com.example.wellbees_project.allSources.SourcesReply
import com.tuann.floatingactionbuttonexpandable.FloatingActionButtonExpandable
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {
    private var sourcesDAOInterface: SourcesDAOInterface? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val recyclerView = findViewById<View>(R.id.source_recyclerview) as RecyclerView
        recyclerView.layoutManager = LinearLayoutManager(this)
        val fab = findViewById<FloatingActionButtonExpandable>(R.id.fab)

        sourcesDAOInterface = ApiUtils.sourcesDaoInterface


        setupFab(fab,applicationContext)
        allSources(recyclerView)
    }

    fun allSources(recyclerView: RecyclerView) {
        val call = sourcesDAOInterface!!.allSources(API_KEY).enqueue(object : Callback<SourcesReply?> {
            override fun onResponse(call: Call<SourcesReply?>, response: Response<SourcesReply?>) {

               var sourceList = response.body()!!.sources

                for(a in sourceList!!) {
                    Log.e("**************", "****************")
                    Log.e("source id: ", a.ıd!!)
                    Log.e("source category: ", a.language!!)
                    Log.e("source id: ", a.description!!)
                }

                recyclerView.adapter = AllSourcesAdapter(sourceList,R.layout.list_item,applicationContext)
            }

            override fun onFailure(call: Call<SourcesReply?>, t: Throwable) {
                Log.e("****************", "hata ----------------")
            }
        } )

    }


    private fun setupFab(fab: FloatingActionButtonExpandable, applicationContext: Context) {
        var fabController: Boolean = true

        fab.setOnClickListener {
            if (fabController == true){
                // eğer expand ise açıktır ve açık haline tıklanınca yeni bir sayfaya yönlenmesini sağlıyorum.
                fabController = false
                fab.expand()

                val intent = Intent(applicationContext, BookmarksActivity::class.java)
                intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
                applicationContext.startActivity(intent)


            }else{
                fabController = true
                fab.collapse()
            }

        }
    }

    companion object {
        private val TAG = MainActivity::class.java.simpleName
        const val API_KEY = "d8920f7f20be4311a9d4e2d76dc68139"
        private const val COUNTRY = "us"
    }
}

